src/pcedit/pc.editors.cpp

```diff
@@ -46,88 +46,83 @@ extern short d_rect_index[80];
 extern bool current_file_has_maps;
 bool choice_active[6];
 
-extern short which_pc_displayed;
 sf::Texture button_num_gworld;
 
 extern std::map<eSkill,short> skill_cost;
 extern std::map<eSkill,short> skill_max;
 extern std::map<eSkill,short> skill_g_cost;
 
-static void put_pc_spells(cDialog& me, short store_trait_mode) {
-	store_trait_mode %= 10;
-	
+// display pc spells
+static void put_pc_spells(cDialog& me, short pc_num, short mode) {
 	for(short i = 0; i < 62; i++) {
 		std::string id = "spell" + std::to_string(i + 1);
 		cLed& cur = dynamic_cast<cLed&>(me[id]);
-		if(((store_trait_mode == 0) && univ.party[which_pc_displayed].mage_spells[i]) ||
-			((store_trait_mode == 1) && univ.party[which_pc_displayed].priest_spells[i]))
+		if((mode == 0 && univ.party[pc_num].mage_spells[i]) ||
+			(mode == 1 && univ.party[pc_num].priest_spells[i]))
 			cur.setState(led_red);
 		else cur.setState(led_off);
 	}
 	
-	me["who"].setText(univ.party[which_pc_displayed].name.c_str());
+	me["who"].setText(univ.party[pc_num].name.c_str());
 }
 
-static bool display_pc_event_filter(cDialog& me, std::string item_hit, const short trait_mode) {
-	short pc_num;
-	
-	pc_num = which_pc_displayed;
-	if(item_hit == "done") {
+static bool display_pc_event_filter(cDialog& me, std::string item_hit, short &pc_num, const short mode, bool edit) {
+	if (edit) {
```

Comment:
No, I'm pretty sure this is wrong. Doesn't this save every time the user clicks an LED?